src/options/managed_streams.h

```diff
@@ -0,0 +1,138 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Gereon Kremer
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2021 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * Wrappers to handle memory management of streams.
+ *
+ * This file contains wrappers to handle special cases of managing memory
+ * related to streams stored in options.
+ */
+
+#include "cvc5_public.h"
+
+#ifndef CVC5__OPTIONS__MANAGED_STREAMS_H
+#define CVC5__OPTIONS__MANAGED_STREAMS_H
+
+#include <memory>
+#include <ostream>
+
+#include "options/options_public.h"
+
+namespace cvc5 {
+
+namespace detail {
```

Comment:
Why `detail`? Can this be moved to the cpp file in an anonymous namespace?

Reply:
They are used in `ManagedStream::open()` which uses `if constexpr` with the condition using the template argument of the class. It thus needs to be in the header, and `openOStream` be declared (though not defined) in the header as well.